docs/RFCS/20210722_on_update_and_row_rehoming.md

```diff
@@ -0,0 +1,334 @@
+expression.
+
+Note that this ON UPDATE expression is distinct from a column's DEFAULT
+expression i.e., a user can set different DEFAULT and ON UPDATE expressions. It
+is also not required that a DEFAULT be specified in order to use ON UPDATE; they
+are separate constructs.
+
+### When does ON UPDATE execute?
+
+| Update mechanism         | ON UPDATE applied (Proposal) | Postgres | MySQL |
+|--------------------------|:-----------------:|:-------:|:--------:|
+| UPDATE                   |         ✅         |✅|✅|
+| UPSERT                   |         ✅         |✅|✅|
+| ON UPDATE CASCADE        |         ✅         |✅|❌|
+| Backfilling a new column |         ❌         |❌|❌|
+
+#### UPDATE
+
+UPDATE statements will apply ON UPDATE expressions when an UPDATE is issued that
+does not modify a column with an ON UPDATE. For example, in a table `t(p int, b
+int)` where `b` has `ON UPDATE 100` if an UPDATE is issued that only touches
+`p`, `b` will be set to 100.
+
+#### UPSERT
+
+UPSERT statements follow the UPDATE rules if the row to be UPSERT already
+exists. If the UPSERT creates a new row, however, an ON UPDATE will not be
+applied.
+
+#### ON UPDATE CASCADE
+
+If a row in a table with an ON UPDATE column is modified by an ON UPDATE CASCADE
+operation, the ON UPDATE expression will be applied. For example, in a table
+`t(p int, b int, c int)`, `c` is a foreign key with ON UPDATE CASCADE, and `b` has
+ON UPDATE, the ON UPDATE for `b` will be applied when `c` gets a cascade.
+
+This decision is made primarily in order to be semantically consistent with ON
+UPDATE in Postgres triggers; Postgres ON UPDATE triggers fire when a row is
+modified via an ON UPDATE CASCADE. This decision will become particularly
+relevant if triggers are implemented in Cockroach in the future.
+
+This behavior can also be motivated via the row rehoming use case. If a REGIONAL
+BY ROW table `rt` has a foreign key onto another RBR table `ft`, we'd like rows
+in `rt` to be in the same region as the rows they reference in `ft`. Therefore,
+an update to a key in `ft` should require that the region for referencing rows
+in `rt` be recalculated, leading to rows in `rt` being in the same region as the
+rows they reference in `ft`. See the rehoming section below for more details on
+how ON UPDATE will be used in REGIONAL BY ROW tables.
+
+#### Backfills
+
+In the case of a backfill resulting from adding a column, however, the ON UPDATE
+clause will not be re-evaluated. Applying the ON UPDATE to all rows is likely
+unexpected behavior, particularly when the ON UPDATE captures something unique
+about each row (region, last modified timestamp, etc.).
+
+### Syntax
+
+Since this addition is a constraint on a column, it can be specified at create
+time with the following syntax:
+
+```
+<column_name> <column_type> <other_constraints> ON UPDATE <update_expr>
+```
+
+The ON UPDATE expression must be standalone i.e., it cannot reference other
```

Comment:
Are there other limitations on the types of expressions that are allowed? Do the same expression limitations for computed columns apply here?